### PR TITLE
MinimumRange and MaximumRange to limit the zoom

### PR DESCRIPTION
**Pull request by tclaff**
Reworked the MinimumRange behaviour and added MaximumRange to limit zooming out. (#401)

### CODE REVIEW

File: Source/OxyPlot/Axes/Axis.cs [hunk 1]
```diff
@@ -362,6 +363,11 @@ public bool IsReversed
         public double MaximumPadding { get; set; }
 
         /// <summary>
+        /// Gets or sets the maximum range of the axis. Setting this property ensures that <c>ActualMaximum-ActualMinimum < <c>MaximumRange</c>. The default value is <c>double.PositiveInfinity</c>.
```
Comment (objorke) [inline Source/OxyPlot/Axes/Axis.cs]: &lt; ?

File: Source/OxyPlot/Axes/Axis.cs [hunk 2]
```diff
@@ -373,7 +379,7 @@ public bool IsReversed
         public double MinimumPadding { get; set; }
 
         /// <summary>
-        /// Gets or sets the minimum range of the axis. Setting this property ensures that <c>ActualMaximum-ActualMinimum > MinimumRange</c>. The default value is <c>0</c>.
+        /// Gets or sets the minimum range of the axis. Setting this property ensures that <c>ActualMaximum-ActualMinimum > <c>MinimumRange</c>. The default value is <c>0</c>.
```
Comment (objorke) [inline Source/OxyPlot/Axes/Axis.cs]: Is the ">" valid?

File: Source/OxyPlot/Axes/Axis.cs [hunk 3]
```diff
@@ -718,25 +724,24 @@ public virtual void CoerceActualMaxMin()
...
             }
 
-            // Coerce the minimum range
             var range = this.ActualMaximum - this.ActualMinimum;
+
+            // Coerce the minimum range
             if (range < this.MinimumRange)
+            { 
+                var average = (this.ActualMaximum + this.ActualMinimum) * 0.5;
+                var delta = this.MinimumRange * 0.5;
```
Comment (objorke) [inline Source/OxyPlot/Axes/Axis.cs]: this is changing the current behavior if Minimum or Maximum are set

Comment (objorke) [inline Source/OxyPlot/Axes/Axis.cs]: I think this was fixed recently, check the history

**Comment by objorke** [comment]
Please fix build errors and verify that the "MinimumRange with Minimum" example is still working.

File: Source/OxyPlot/Axes/Axis.cs (hunk 1 above)
Comment (tclaff) [inline Source/OxyPlot/Axes/Axis.cs]: I think it's correct, but the second <c> is not.

File: Source/OxyPlot/Axes/Axis.cs (hunk 2 above)
Comment (tclaff) [inline Source/OxyPlot/Axes/Axis.cs]: I think it is correct.

File: Source/OxyPlot/Axes/Axis.cs (hunk 3 above)
Comment (tclaff) [inline Source/OxyPlot/Axes/Axis.cs]: This explains what the code was intended for. However, i think the fix for #550 broke the intended feature of the MinimumRange property. In the example "MinimumRange with Minimum" the MinimumRange is set to 5 but the user can clearly zoom beyond 5 units on the axis. I am not sure what to do about this.

**Comment by tclaff** [comment]
I have reverted the MinimumRange behavior to the old one. Since i think there is an issue with it, i will open a separate ticket for that.

**Comment by objorke** [comment]
You are right, something is wrong with the MinimumRange implementation. #550

**Pull request merged**